Makefile

```diff
@@ -6,29 +6,54 @@ K8S_VERSION_MINOR := $(word 1,${K8S_VERSION_PARTS}).$(word 2,${K8S_VERSION_PARTS
 
 MAKEFILE_DIR := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
 
+# expands to 'true' if PACKER_VARIABLE_FILE is non-empty
+# and the file contains the string passed as the first argument
+# otherwise, expands to 'false'
+packer_variable_file_contains = $(if $(PACKER_VARIABLE_FILE),$(shell grep -Fq $1 $(PACKER_VARIABLE_FILE) && echo true || echo false),false)
+
+# expands to 'true' if the version comparison is affirmative
+# otherwise expands to 'false'
+vercmp = $(shell $(MAKEFILE_DIR)/files/bin/vercmp "$1" "$2" "$3")
+
+# expands to 'true' if the 'aws_region' contains 'us-iso' (an isolated region)
+# otherwise, expands to 'false'
+in_iso_region = $(if $(findstring us-iso,$(aws_region)),true,false)
+
+# gp3 volumes are used by default for 1.27+
+# TODO: remove when 1.26 reaches EOL
+# TODO: remove when gp3 is supported in isolated regions
+ifneq ($(call packer_variable_file_contains,volume_type), true)
+	ifeq ($(call in_iso_region), true)
+		volume_type ?= gp2
+	else ifeq ($(call vercmp,$(kubernetes_version),lt,1.27.0), true)
+		volume_type ?= gp2
+	endif
```

Comment:
`make` has an `or` function, but it treats the empty string as `false` and any non-empty string as `true`. That's easy enough to work with, but it makes the `vercmp` and `in_iso_region` a bit clunkier to work with in general, and the resulting `or` is just a longer, less readable line. 🤷

Reply:
Not totally following here. Are you explaining why you have two `if`s that result in the first action versus combining these checks into one line?

Reply:
Yeah, the repetition of `volume_type ?= gp2` isn't ideal, but this is nicer than the alternative